diff --git a/meson.build b/meson.build
index 74fa82a6..26152229 100644
--- a/meson.build
+++ b/meson.build
@@ -403,7 +403,7 @@ libinput_version_h = configure_file(
 
 mapfile = dir_src / 'libinput.sym'
 version_flag = '-Wl,--version-script,@0@'.format(mapfile)
-lib_libinput = shared_library('input',
+lib_libinput = library('input',
 		src_libinput,
 		include_directories : [include_directories('.'), includes_include],
 		dependencies : deps_libinput,
